server/src/main/java/org/elasticsearch/bootstrap/ESPolicy.java

```diff
@@ -166,7 +166,7 @@ private boolean canAccessSecuredFile(URL location, FilePermission permission) {
         Set<URL> accessibleSources = securedFiles.get(permission);
         if (accessibleSources != null) {
             // simple case - single-file referenced directly
-            return accessibleSources.contains(location);
+            return accessibleSources.contains(location) || location.toString().equals("jrt:/java.security.jgss");
```

Comment:
Finally managed to figure out the necessary bit to let Kerberos authentication work with enhanced protections on the keytab file - obviously we can't merge this hack as-is, this is just for demonstrative purposes, but @rjernst or @thecoop do you have advice for the best way to accomplish this in a less hacky way?

Reply:
Can we not do a grant for this?

Reply:
@rjernst Probably should have mentioned this up front, but I thought it might be a known limitation. If I add:
```
grant codebase "jrt:/java.security.jgss" {
    permission org.elasticsearch.SecuredConfigFileSettingAccessPermission "xpack.security.authc.realms.kerberos.*.keytab.path";
  }
``` 

to Security's policy file, none of the secured config file permissions are loaded correctly. As in, none of them, across all plugins, silently. I'm still digging into why, but this is pretty unfamiliar code.

Reply:
That actually appears to be true for any codebase grant of `SecuredConfigFileSettingAccessPermission`, not just the `jrt:` URL.

Reply:
codebase must be a url, but I guess jrt is not handled (normally we plug in a jar file path). On the policy file silently ignoring things, this is a known (and horrendous) behavior of Java's policy parsing.

Reply:
I spent some time going through the relevant code to see if I could figure out where the loss was happening and did not really manage to. I did figure out that `java.security.jgss` is one of a few things granted `AllPermission` by the JVM, so I replaced the hardcoded check for that module specifically with a check to allow code with `AllPermission` to access secured files. I’m not sure that’s what we want, but it’s at least less hacky.

Reply:
That seems like a more durable solution. We don't allow AllPermission in our plugin policy files, so if the JDK is using it, allowing it seems reasonable.